runners/spark/src/main/java/org/apache/beam/runners/spark/translation/DoFnFunction.java

```diff
@@ -61,8 +65,20 @@ public DoFnFunction(DoFn<InputT, OutputT> fn,
       Exception {
     ProcCtxt ctxt = new ProcCtxt(mFunction, mRuntimeContext, mSideInputs);
     ctxt.setup();
-    mFunction.startBundle(ctxt);
-    return ctxt.getOutputIterable(iter, mFunction);
+    try {
+      mFunction.setup();
+      mFunction.startBundle(ctxt);
+      return ctxt.getOutputIterable(iter, mFunction);
+    } catch (Exception e) {
+      try {
+        mFunction.teardown();
+      } catch (Exception teardownException) {
+        LOG.error(
```

Comment:
I'm not sure what additional logging occurs; there will not be any messaging around the "teardown while trying to clean up after setup/startBundle threw an exception" as far as I know